### PR TITLE
feat(assembly): add Mix umbrella topology assembly

### PR DESCRIPTION
**Pull request by mstykow**
## Summary

- Add a `TopologyPlan` domain (`src/assembly/mix_umbrella_merge.rs`) for Elixir Mix **umbrella** projects: a root `mix.exs` with `apps_path:` and one or more child apps under that directory now assemble as a coherent multi-package domain instead of sibling-local merge only.
- Each recognized child app becomes its own `pkg:hex/<app>@<version>` package. The umbrella's single shared root `mix.lock` is attributed per-app based on which app's own `mix.exs` directly declares that locked dependency name; an entry no member declares stays honestly hoisted (`for_package_uid: None`), matching the existing standalone-`mix.lock` fallback.
- A dependency declared `{:sibling, in_umbrella: true}` (`src/parsers/mix_exs.rs`) is resolved to the sibling app's real, versioned purl instead of a fabricated unversioned `pkg:hex/sibling`; a dangling reference (filtered out by `apps:`, or nonexistent) is dropped rather than guessed.
- An umbrella root with no `app:` (the common case — an umbrella root usually isn't its own OTP application) forms no package of its own; its own deps and any lock entries no member claims stay hoisted, mirroring the Cargo/npm workspace-root-without-its-own-manifest fallback. A root that *does* have `app:` is treated as another owning candidate, like a member.
- A directory under `apps_path` that isn't a recognized member (excluded by an `apps:` allow-list, or missing a valid `mix.exs`) is left untouched by umbrella assembly rather than swept into the root/all-members file-ownership fallback.

## Issues

- Covers: umbrella `apps_path` trees previously breaking CycloneDX/SPDX graphs by either merging into one package or leaving `mix.lock` deps unowned.
- Closes: N/A

## Scope and exclusions

- Included: Mix/Hex umbrella topology planning and assembly wiring only (one ecosystem family).
- Explicit exclusions: no changes to SBOM writers (CycloneDX/SPDX); the existing `OutputPackage`/`OutputTopLevelDependency` schema already renders the new multi-package/attributed-dependency shape correctly, proven by the new assembly golden below. No changes to `docs/SUPPORTED_FORMATS.md` (no parser metadata changed).

## How to verify

- `cargo test --lib mix` — parser unit tests, umbrella-specific parser tests, and a full scan→assemble Layer 3 test (`mix_exs_scan_test.rs`) covering the two-app umbrella + shared lock + `in_umbrella` resolution end to end.
- `cargo test --lib assembly::` — includes two new focused unit tests for edge cases: a dangling `in_umbrella` reference being dropped, and an `apps:`-excluded app neither absorbing shared-lock ownership nor being swept into the fallback file attribution.
- `cargo test --features golden-tests --test assembly_golden test_assembly_mix_umbrella` — new fixture-backed golden (`testdata/assembly-golden/mix-umbrella/`) asserting the full two-package/unique-deps/file-ownership shape.
- `cargo clippy --lib --tests -- -D warnings` and `cargo fmt --check` on touched files — clean.

## Intentional differences from Python

- ScanCode does not model Mix umbrella topology as a multi-package domain at all; Provenant's per-app lock attribution (based on each app's own direct deps) and honest hoisting of unclaimed/root-hoisted lock entries has no direct ScanCode analog to diverge from.

## Follow-up work

- Created or intentionally deferred: none identified; nested umbrellas-of-umbrellas and non-standard `apps_path` layouts are out of scope and fall back to the existing standalone/sibling-merge behavior.

## Expected-output fixture changes

- Files changed: `testdata/assembly-golden/mix-umbrella/expected.json` (new fixture, not a modification of an existing one).
- Why the new expected output is correct: generated by the golden harness from the new fixture, then reviewed by hand to confirm two packages (`app_one@0.1.0`, `app_two@0.2.0`), the `in_umbrella` edge resolved to `app_two`'s real identity, `phoenix` attributed to `app_one` and `ecto` to `app_two` from the shared `mix.lock`, and both root files (`mix.exs`, `mix.lock`) linked to both member packages.

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: src/assembly/topology.rs [hunk 1]
```diff
@@ -227,6 +264,10 @@ impl TopologyPlan {
             return self.claimed_hackage_dirs.contains(parent_dir);
         }
 
+        if config.datasource_ids.contains(&DatasourceId::HexMixExs) {
+            return self.claimed_mix_dirs.contains(parent_dir);
+        }
```
[the remark this answers is not in the document]
Comment (mstykow) [reply src/assembly/topology.rs:269]: Fixed in e518a94e34b3d74a15f8f2834b568ebb96a8dbc8.

Chosen contract: a member directory is still claimed for umbrella topology (not sibling-merged normally), but a mix.lock living directly inside that member's own directory is now merged in addition to the umbrella-wide root mix.lock attribution. Since a member-local lock is unambiguously that member's own (unlike the shared root lock, which needs direct-dep-name attribution to figure out ownership), every entry in it is attributed straight to that member — matching the existing standalone mix.exs+mix.lock sibling-merge contract.

Added a golden fixture (testdata/assembly-golden/mix-umbrella-member-lock/) plus focused assembly (assembly_test.rs) and scan-level (mix_exs_scan_test.rs) tests covering a member with its own mix.lock, and updated the module contract doc comment in mix_umbrella_merge.rs.

@greptileai review
